### PR TITLE
qgs3dmapcanvaswidget: Allow to draw the extent on the 2D map canvas

### PR DESCRIPTION
**Pull request by ptitjano**
This adds a QgsRubberband in the 2D map canvas similar to the one used to display the Frustum.

The option can be enabled in the General section of the 3D configuration widget.

The option 
Capture d’écran du 2023-03-07 12-41-20

Result on the 2D Map Canvas

Capture d’écran du 2023-03-07 12-41-29

Extent and Frustum combined

Capture d’écran du 2023-03-07 12-41-59

cc @benoitdm-oslandia @wonder-sk

### CODE REVIEW

File: src/3d/qgs3dmapsettings.cpp [hunk 1]
```diff
@@ -141,6 +142,8 @@ void Qgs3DMapSettings::readXml( const QDomElement &elem, const QgsReadWriteConte
                 elemExtent.attribute( QStringLiteral( "yMin" ) ).toDouble(),
                 elemExtent.attribute( QStringLiteral( "xMax" ) ).toDouble(),
                 elemExtent.attribute( QStringLiteral( "yMax" ) ).toDouble() );
+
+    mShowExtentIn2DView = elemExtent.attribute( QStringLiteral( "show-in-2d-view" ), QStringLiteral( "0" ) ).toInt();
```
Comment (nyalldawson) [inline src/3d/qgs3dmapsettings.cpp:146]:
```suggestion
    mShowExtentIn2DView = elemExtent.attribute( QStringLiteral( "showIn2dView" ), QStringLiteral( "0" ) ).toInt();
```

There's a lot of exceptions, but predominantly camel case is used in the QGIS xml elements.

File: src/3d/qgs3dmapsettings.h [hunk 2]
```diff
@@ -681,6 +681,19 @@ class _3D_EXPORT Qgs3DMapSettings : public QObject, public QgsTemporalRangeObjec
...
+     * \see setShowExtentIn2DView()
+     * \since QGIS 3.32
+     */
+    bool showExtentIn2DView() const { return mShowExtentIn2DView; }
+
+    /**
+     * Sets whether the extent is displayed on the main 2D map canvas
+     * \since QGIS 3.32
+     */
+    void setShowExtentIn2DView( bool showExtentIn2DViewEnabled );
```
Comment (nyalldawson) [inline src/3d/qgs3dmapsettings.h:695]:
```suggestion
    void setShowExtentIn2DView( bool show );
```

File: src/app/3d/qgs3dmapcanvaswidget.h [hunk 3]
```diff
@@ -117,6 +118,7 @@ class APP_EXPORT Qgs3DMapCanvasWidget : public QWidget
     QToolButton *mBtnOptions = nullptr;
     QgsDockableWidgetHelper *mDockableWidgetHelper = nullptr;
     QgsRubberBand *mViewFrustumHighlight = nullptr;
+    QgsRubberBand *mViewExtentHighlight = nullptr;
```
Comment (nyalldawson) [inline src/app/3d/qgs3dmapcanvaswidget.h:121]:
```suggestion
    QObjectUniquePtr< QgsRubberBand > mViewExtentHighlight;
```

likely is safer

File: src/app/3d/qgs3dmapconfigwidget.cpp [hunk 4]
```diff
@@ -252,6 +252,11 @@ Qgs3DMapConfigWidget::Qgs3DMapConfigWidget( Qgs3DMapSettings *map, QgsMapCanvas
   groupExtent->setCurrentExtent( mMap->extent(), mMap->crs() );
   groupExtent->setOutputExtentFromCurrent();
   groupExtent->setMapCanvas( mMainCanvas );
+
+  // checkbox to display the extent in the 2D Map View
+  mShowExtentIn2DViewCheckbox = new QCheckBox( tr( "Show in 2D Map View" ) );
```
Comment (nyalldawson) [inline src/app/3d/qgs3dmapconfigwidget.cpp:257]:
```suggestion
  mShowExtentIn2DViewCheckbox = new QCheckBox( tr( "Show in 2D map view" ) );
```

File: src/3d/qgs3dmapsettings.cpp (hunk 1 above)
Comment (ptitjano) [reply src/3d/qgs3dmapsettings.cpp]: Indeed there are a lot of exceptions! Done.

File: src/3d/qgs3dmapsettings.h (hunk 2 above)
Comment (ptitjano) [reply src/3d/qgs3dmapsettings.h]: Done.

File: src/app/3d/qgs3dmapcanvaswidget.h (hunk 3 above)
Comment (ptitjano) [reply src/app/3d/qgs3dmapcanvaswidget.h]: Done. I have also added a commit to change it for `mViewFrustumHighlight`.

File: src/app/3d/qgs3dmapconfigwidget.cpp (hunk 4 above)
Comment (ptitjano) [reply src/app/3d/qgs3dmapconfigwidget.cpp]: Done. I have also added an other commit to change it for some other labels.

File: src/app/3d/qgs3dmapcanvaswidget.cpp [hunk 5]
```diff
@@ -167,7 +166,7 @@ Qgs3DMapCanvasWidget::Qgs3DMapCanvasWidget( const QString &name, bool isDocked )
 
   mOptionsMenu->addSeparator();
 
-  mActionSync2DNavTo3D = new QAction( tr( "2D Map View Follows 3D Camera" ), this );
+  mActionSync2DNavTo3D = new QAction( tr( "2D map view follows 3D camera" ), this );
```
Comment (nyalldawson) [inline src/app/3d/qgs3dmapcanvaswidget.cpp:169]: Actually actions SHOULD have title case -- it's just checkboxes and label widgets which need sentence case!

Comment (ptitjano) [reply src/app/3d/qgs3dmapcanvaswidget.cpp]: My bad. This is fixed.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**